lib/req/steps.ex

```diff
@@ -374,6 +374,26 @@ defmodule Req.Steps do
     end
   end
 
+  @doc """
+  Compresses the request body.
+
+  ## Request options
+
+    * `:compress_request` - compresses the request body using `gzip` compression and sets the
+    `"Content-Encoding: gzip"` header. Defaults to `false`.
+
+  """
+  @doc step: :request
+  def compress_request(request) do
```

Comment:
Given we have `decompress` response step, let's call this one `compress`.

If you think a more verbose name would be better, let's call this `compress_body` then? (Since we have `decompress_body` response step.)

```suggestion
  def compress(request) do
```

Reply:
Let's do `compress_body` if you agree.. Having `compress` and `compressed` is a big opportunity for confusion. It also makes it hard to search through the code.

Reply:
If we don’t have compressed would you be ok with calling it just compress? I agree about possible confusion

Reply:
I suppose it depends on what we have instead but I assume it would.

Reply:
@srowley sorry for back'n'forth and bike shedding. Let's go with your idea of `compress_body/1` and `:compress_body` option.

I tend to prefer shorter option names but this would be rather seldom used one so it's ok if it's a bit longer. It's definitely super clear. I'll probably keep the name `compressed` though since there's `curl --compressed` and can't think of a better one.

If you'd like to send a PR to rename decompress to decompress_body that would be appreciated too but no worries, happy to do it myself once this lands. Thank you.